basic-api.yml

```diff
@@ -0,0 +1,142 @@
+    "@context": [
+      "https://www.w3.org/2018/credentials/v1",
+      "https://w3id.org/citizenship/v1"
+    ],
+    "id": "https://issuer.oidp.uscis.gov/credentials/83627465",
+    "type": ["VerifiableCredential", "PermanentResidentCard"],
+    "issuer": "did:v1:test:nym:z6Mko5ZVKPDsuTQN...46ecbs1u75hk2RJgAP7XU",
+    "issuanceDate": "2019-12-03T12:19:52Z",
+    "expirationDate": "2029-12-03T12:19:52Z",
+    "credentialSubject": {
+      "id": "did:example:b34ca6cd37bbf23",
+      "type": ["PermanentResident", "Person"],
+      "givenName": "JOHN",
+      "familyName": "SMITH",
+      "gender": "Male",
+      "image": "data:image/png;base64,iVBORw0KGgo...kJggg==",
+      "residentSince": "2015-01-01",
+      "lprCategory": "C09",
+      "lprNumber": "999-999-999",
+      "commuterClassification": "C1",
+      "birthCountry": "Bahamas",
+      "birthDate": "1958-07-17"
+    }
+  }],
+  "proof": [{
+    "type": "Ed25519Signature2018",
+    "created": "2020-02-25T22:03:05Z",
+    "jws": "eyJhbGciOiJFZERTQSIsImI2..hS5SPmWTFocjcDg",
+    "proofPurpose":"assertionMethod",
+    "verificationMethod":"did:key:z6Mkm8aUytv...nvKQrHB"
+  }]
+}
+        '400':
+          description: Malformed request from client.
+components:
+  schemas:
+    IssueRequest:
+      type: object
+      properties:
+        credential:
+          type: object
+          description: The credential to be issued minus the proofs
+          example: {
+  "@context": [
+    "https://www.w3.org/2018/credentials/v1",
+    "https://w3id.org/citizenship/v1"
+  ],
+  "id": "https://issuer.oidp.uscis.gov/credentials/83627465",
+  "type": [
+    "VerifiableCredential",
+    "PermanentResidentCard"
+  ],
+  "issuanceDate": "2019-12-03T12:19:52Z",
+  "expirationDate": "2029-12-03T12:19:52Z",
+  "credentialSubject": {
+    "id": "did:example:b34ca6cd37bbf23",
+    "type": [
+      "PermanentResident",
+      "Person"
+    ],
+    "givenName": "JOHN",
+    "familyName": "SMITH",
+    "gender": "Male",
+    "image": "data:image/png;base64,iVBORw0KGgo...kJggg==",
+    "residentSince": "2015-01-01",
+    "lprCategory": "C09",
+    "lprNumber": "999-999-999",
+    "commuterClassification": "C1",
+    "birthCountry": "Bahamas",
+    "birthDate": "1958-07-17"
+  }
+}
+        options:
+          type: object
+          description: Optional values that check or modify the input and modify the output of the request.
+          properties:
+            validation:
+              type: object
+              description: Optional validation schemas to use on the input document.
+              properties:
+                type:
+                  type: string
+                  description: The type of validation mechanism
+                  example: JsonSchema2018
+                credentialSchema:
+                  type: string
+                  description: A link to a credential schema to use for validation.
+                  example: https://example.com/schemas/prc.json
+            issuer:
```

Comment:
How do we specify the keyID?

Reply:
hopefully something like `did:example:123#fingerprint`... or `fingerprint`.

Reply:
How this works is certainly missing from the API/documentation.

These API changes imply that there should be some kind of "issuer instance" that includes state that is configured elsewhere that would include this kind of information. This includes information such as which types of credentials an "issuer instance" (or a particular client of that instance) may issue and which keys may be used to sign those different types of credentials. The idea is that the caller of the API does not necessarily care about every layer involved in the issuing process -- and they may not even be authorized to make those sorts of decisions. I think we need to start capturing this concept somewhere and potentially exposing more API surface for it.

Another form of information we would want to be able to store with an issuer instance (or a verifier instance for that matter) would be the JSON-LD contexts that are supported. There should be an API endpoint for populating an instance with the contexts it is authorized to use.

None of this precludes an option to allow a key ID to be specified as an override; if it is acceptable to that "issuer instance".

Reply:
While I agree with what @dlongley says above, I don't think now is the time to do that. Let's focus on the *one* API that gets us to interop... everything else is icing on the cake.

@llorllale - you're right, it's missing from the API. I'll add a way to specify the key that should generate the proof for the VC. Having an endpoint that "knows" what the key should be (as @dlongley mentions above) is a nice to have, not required for this minimal API (that is, we can add it in a separate PR).

Reply:
@msporny Thank you. I think what is paramount is clarifying the model for everyone involved. @dlongley points out some possible assumptions behind the design of this API. We should in parallel be clarifying the capabilities we need this service to have, and this API addresses that concern only in part.